test/messaging.go

```diff
@@ -22,15 +33,21 @@ type MessagingClient struct {
 }
 
 // NewMessagingClient returns a new instance of MessagingClient.
-func NewMessagingClient() *MessagingClient {
+func NewMessagingClient(dataURL url.URL) *MessagingClient {
 	c := &MessagingClient{
 		messagesByTopicID: make(map[uint64][]*messaging.Message),
+		dataURL:           dataURL,
 	}
 	c.PublishFunc = c.DefaultPublishFunc
 	c.ConnFunc = c.DefaultConnFunc
 	return c
 }
 
+// NewMessagingClient returns a new instance of MessagingClient.
+func NewTestMessagingClient() *MessagingClient {
+	return NewMessagingClient(*testDataURL)
+}
```

Comment:
Can we change the name of this to `NewDefaultMessagingClient()`? Otherwise it creates a stutter in the caller code: `test.NewTestMessagingClient()` and it's not easy to understand how it's different from the `test.NewMessagingClient()`.